From e05e7241b223de95c0a08c78ffd47c8a332f826d Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:29:35 -0400
Subject: [alsa] hda/realtek: Improve HP depop when system change power state on Chromebook

Message-id: <140972823573-256-patchset-jkysela@redhat.com>
Patchwork-id: 90956
O-Subject: [RHEL 7.1 PATCH 255/442] ALSA: hda/realtek - Improve HP depop when system change power state on Chromebook
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

It is better to change Mic2-Vref to manual mode.
Manual control Mic2-Vref will solve pop noise issue.
It will improve pop noise for power on, power off, S3 and resume.

Signed-off-by: Kailang Yang <kailang@realtek.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: de9481cb40339d9bfc1104b53649876fc5f3d432)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 1dafaa9..b6ec268 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -4005,6 +4005,10 @@ static void alc283_fixup_chromebook(struct hda_codec *codec,
   spec->gen.mixer_nid = 0;
   break;
  case HDA_FIXUP_ACT_INIT:
+  /* MIC2-VREF control */
+  /* Set to manual mode */
+  val = alc_read_coef_idx(codec, 0x06);
+  alc_write_coef_idx(codec, 0x06, val & ~0x000c);
   /* Enable Line1 input control by verb */
   val = alc_read_coef_idx(codec, 0x1a);
   alc_write_coef_idx(codec, 0x1a, val | (1 << 4));
-- 
1.7.1